internal/envelope/envelope.go

```diff
@@ -0,0 +1,39 @@
+package envelope
+
+import (
+	"errors"
+
+	"github.com/notaryproject/notation-core-go/signature/cose"
+	"github.com/notaryproject/notation-core-go/signature/jws"
+	gcose "github.com/veraison/go-cose"
+)
+
+// Supported envelope format.
+const (
+	COSE = "cose"
+	JWS  = "jws"
+)
+
+// SpeculateSignatureEnvelopeFormat speculates envelope format by looping all builtin envelope format.
+//
+// TODO: do we nned to speculate evenlope format?
+//
+// Both push and verify will need to speculate format now not only for caching.
+//
+// For verifying, the new verification workflow with trust store
+// and trust policy doesn't need to speculate format because it will pull from remote.
+//
+// For pushing, there is a flag named signatures, which is the paths to some signatures. CLI will first read the file and push its content to the remote.
+//
+// For RC1, verify doesn't need speculate, but we still need to speculate format if user pushes a signature from local file system
```

Comment:
Please like issue https://github.com/notaryproject/notation/issues/379 and remove the description here and add it to issue https://github.com/notaryproject/notation/issues/379

As per discussion with Yi we are getting rid of signature cache for RC1.

Reply:
Add comments to #379